crates/compiler/parse/src/expr.rs

```diff
@@ -264,6 +264,8 @@ fn crash_kw<'a>() -> impl Parser<'a, Expr<'a>, EExpr<'a>> {
     }
 }
 
+// avoids ownership issues
+#[allow(clippy::blocks_in_conditions)]
```

Comment:
I think it's better to fix the actual lint error.

```
diff --git a/crates/compiler/parse/src/expr.rs b/crates/compiler/parse/src/expr.rs
index 0fe317ad16..50ca70fa2e 100644
--- a/crates/compiler/parse/src/expr.rs
+++ b/crates/compiler/parse/src/expr.rs
@@ -265,21 +265,21 @@ fn crash_kw<'a>() -> impl Parser<'a, Expr<'a>, EExpr<'a>> {
 }
 
 // avoids ownership issues
-#[allow(clippy::blocks_in_conditions)]
 fn loc_possibly_negative_or_negated_term<'a>(
     options: ExprParseOptions,
 ) -> impl Parser<'a, Loc<Expr<'a>>, EExpr<'a>> {
-    one_of![
-        |arena, state: State<'a>, min_indent: u32| {
-            let initial = state.clone();
+    let parse_unary_negate = move |arena, state: State<'a>, min_indent: u32| {
+        let initial = state.clone();
 
-            let (_, (loc_op, loc_expr), state) =
-                and!(loc!(unary_negate()), loc_term(options)).parse(arena, state, min_indent)?;
+        let (_, (loc_op, loc_expr), state) =
+            and!(loc!(unary_negate()), loc_term(options)).parse(arena, state, min_indent)?;
 
-            let loc_expr = numeric_negate_expression(arena, initial, loc_op, loc_expr, &[]);
+        let loc_expr = numeric_negate_expression(arena, initial, loc_op, loc_expr, &[]);
 
-            Ok((MadeProgress, loc_expr, state))
-        },
+        Ok((MadeProgress, loc_expr, state))
+    };
+    one_of![
+        parse_unary_negate,
         // this will parse negative numbers, which the unary negate thing up top doesn't (for now)
         loc!(specialize_err(EExpr::Number, number_literal_help())),
         loc!(map_with_arena!(
```